src/desktop/layer.rs

```diff
@@ -587,24 +623,22 @@ impl LayerSurface {
     /// Subsequent calls will return an empty vector until the buffer is updated again.
     pub(super) fn accumulated_damage(
         &self,
+        scale: impl Into<Scale<f64>>,
         for_values: Option<(&Space, &Output)>,
-    ) -> Vec<Rectangle<i32, Logical>> {
+    ) -> Vec<Rectangle<i32, Physical>> {
+        let scale = scale.into();
         let mut damage = Vec::new();
         if let Some(surface) = self.get_surface() {
-            damage.extend(
-                damage_from_surface_tree(surface, (0, 0), for_values)
-                    .into_iter()
-                    .flat_map(|rect| rect.intersection(self.bbox())),
```

Comment:
Yes, the src should always be defined. If we have a buffer we also calculate the src, in case no viewport has been defined it corresponds to the buffer logical size.